tests/t567-operator.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+CEED_QFUNCTION(setup)(void *ctx, const CeedInt Q,
+                      const CeedScalar *const *in,
+                      CeedScalar *const *out) {
+    // *INDENT-OFF*
+  const CeedScalar                  *w = in[0],
+                   (*J)[2][CEED_Q_VLA] = (const CeedScalar(*)[2][CEED_Q_VLA])in[1];
+  CeedScalar     (*q_data)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0];
+    // *INDENT-ON*
+
+  // Quadrature point loop
+  CeedPragmaSIMD
+  for (CeedInt i=0; i<Q; i++) {
+    // Qdata stored in Voigt convention
+    // J: 0 2   q_data: 0 2   adj(J):  J11 -J01
+    //    1 3           2 1           -J10  J00
+    const CeedScalar J00 = J[0][0][i];
+    const CeedScalar J10 = J[0][1][i];
+    const CeedScalar J01 = J[1][0][i];
+    const CeedScalar J11 = J[1][1][i];
+    const CeedScalar qw = w[i] / (J00*J11 - J10*J01);
+    q_data[0][i] =   qw * (J01*J01 + J11*J11);
+    q_data[1][i] =   qw * (J00*J00 + J10*J10);
+    q_data[2][i] = - qw * (J00*J01 + J10*J11);
+  } // End of Quadrature Point Loop
+
+  return 0;
+}
+
+CEED_QFUNCTION(diff)(void *ctx, const CeedInt Q, const CeedScalar *const *in,
+                     CeedScalar *const *out) {
+    // *INDENT-OFF*
+  const CeedScalar (*q_data)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[0],
+                       (*ug)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[1];
+  CeedScalar           (*vg)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0];
+    // *INDENT-ON*
+
+  const CeedInt dim = 2;
+  const CeedScalar scale[2][2] = {
+    {1.0, 2.0},
+    {3.0, 4.0},
+  };
+
+  // Quadrature point loop
+  CeedPragmaSIMD
+  for (CeedInt i=0; i<Q; i++) {
+    // Read qdata (dXdxdXdxT symmetric matrix)
+    // Stored in Voigt convention
+    // 0 2
+    // 2 1
+    // *INDENT-OFF*
+    const CeedScalar dXdxdXdxT[2][2] = {{q_data[0][i],
+                                         q_data[2][i]},
+                                        {q_data[2][i],
+                                         q_data[1][i]}
+                                       };
+    // *INDENT-ON*
+
+    // Apply Poisson operator
+    // j = direction of vg
+    for (CeedInt j=0; j<dim; j++)
+      vg[j][i] = (ug[0][i] * dXdxdXdxT[0][j] * scale[0][i] +
+                  ug[1][i] * dXdxdXdxT[1][j] * scale[1][i]);
```

Comment:
I think at least some of the current issues with this test are from this line -- `i` is the quadrature point loop, so I don't think it's what you intended for indexing the scale values, especially since Q is 3 and the size of scale is 2x2.

Switching `scale[*][i]` to `scale[*][j]` here fixes all the issues I was seeing locally (avx/blocked, hip-ref) but not sure if that's what you intended, either.  With the modified QFunction, I confirmed that the `values` array from the assembly was the same for cpu and hip-ref.

Reply:
Good catch. I'm not if this is good or bad news for the fluids assembly bug.

Reply:
Hmm, yeah, though I don't know why, since they use the same (ref) kernels for assembly...

Reply:
It seems to be the same error between hip-gen and cuda-gen.

Reply:
If I'm understanding the output correctly, the `assembled_true_value` is the one that is wrong.

Reply:
Yeah, I've checked, and the output from the setup operator is the same, but then the outputs from the apply operator are wrong for cuda-gen, but not cuda-ref. ??  Been wasting a bunch of time trying to track down the reason, but no real good theories yet.

Reply:
Okay I think I found the issue.  The QFunction for the application in t567 doesn't include the multiple components anywhere, unlike t566-operator.h

Reply:
Is there something like valgrind client requests? Or we could make libCEED device malloc initialize with NaN.